custom/icds_reports/static/js/directives/indie-map/indie-map.directive.js

```diff
@@ -163,6 +163,19 @@ function IndieMapController($scope, $compile, $location, $filter, storageService
                     projection.scale(scaleCenter.scale)
                         .center(scaleCenter.center)
                         .translate([element.offsetWidth / 2, element.offsetHeight / div]);
+
+                    //setting zoom out limit
+                    //references: https://data-map-d3.readthedocs.io/en/latest/steps/step_06.html#step-06
+                    //            bower_components/angular-datamaps/dist/angular-datamaps.js (line 36)
+                    $(function(){
+                        // DOM Ready
+                        var svg = d3.select('#map svg');
+                        var zoom = d3.behavior.zoom().scaleExtent([1, 10]).on('zoom', function() {
+                          svg.selectAll('g').attr("transform",
+                            "translate(" + d3.event.translate + ") scale(" + d3.event.scale + ")");
```

Comment:
can you explain what this does?

Reply:
Datamaps is svg-based map-rendering library. So we are selecting "svg"s after map is rendered and adding an event listener ("zoom") to it with scaling range.
```suggestion
svg.selectAll('g').attr("transform",
                            "translate(" + d3.event.translate + ") scale(" + d3.event.scale + ")");
```
This redraws the map every time zoom event triggers. (Reference: library file (angular-datamaps.js))

```suggestion
svg.call(zoom);
```
This connects zoom event to svg

Reply:
Datamaps is svg-based map-rendering library. So we are selecting "svg"s after map is rendered and adding an event listener ("zoom") to it with scaling range.

```
svg.selectAll('g').attr("transform",
                            "translate(" + d3.event.translate + ") scale(" + d3.event.scale + ")");

```
This redraws the map every time zoom event triggers. (Reference: library file (angular-datamaps.js))

```
svg.call(zoom);
```
This connects zoom event to svg